src/common/providers/__tests__/TranslatableProvider.test.tsx

```diff
@@ -0,0 +1,132 @@
+vi.mock('../components/languageTab/LanguageTabs', () => {
+  return function MockLanguageTabs() {
+    return <div data-testid="mock-language-tabs">Mock Language Tabs</div>;
+  };
+});
+
+describe('TranslatableProvider', () => {
+  it('renders children within the provider', () => {
+    render(
+      <TranslatableProvider>
+        <div data-testid="child-element">Child Element</div>
+      </TranslatableProvider>
+    );
+
+    expect(screen.getByTestId('child-element')).toBeInTheDocument();
+  });
+
+  it('provides the correct context values', () => {
+    let contextValue: any;
+    render(
+      <TranslatableProvider>
+        <TranslatableContext.Consumer>
+          {(value) => {
+            contextValue = value;
+            return null;
+          }}
+        </TranslatableContext.Consumer>
+      </TranslatableProvider>
+    );
+
+    expect(contextValue).toBeDefined();
+    expect(contextValue.getSource).toBeInstanceOf(Function);
+    expect(contextValue.defaultLanguage).toBe(Language.Fi);
+    expect(contextValue.languages).toEqual(CONTENT_LANGUAGES);
+    expect(contextValue.selectLanguage).toBeInstanceOf(Function);
+    expect(contextValue.selectedLanguage).toBe(Language.Fi);
+    expect(contextValue.selector).toBeDefined();
+  });
+
+  it('uses the correct default values when not specified', () => {
+    let contextValue: any;
+    render(
+      <TranslatableProvider>
+        <TranslatableContext.Consumer>
+          {(value) => {
+            contextValue = value;
+            return null;
+          }}
+        </TranslatableContext.Consumer>
+      </TranslatableProvider>
+    );
+
+    expect(contextValue.defaultLanguage).toBe(Language.Fi);
+    expect(contextValue.languages).toEqual(CONTENT_LANGUAGES);
+  });
+
+  it('overrides default language, languages and selector when provided', () => {
+    const customLanguages = [Language.En, Language.Sv];
+    const customDefaultLanguage = Language.En;
+    const customSelector = (
+      <div data-testid="custom-selector">Custom Selector</div>
+    );
+
+    let contextValue: any;
+    render(
+      <TranslatableProvider
+        defaultLanguage={customDefaultLanguage}
+        languages={customLanguages}
+        selector={customSelector}
+      >
+        <TranslatableContext.Consumer>
+          {(value) => {
+            contextValue = value;
+            return null;
+          }}
+        </TranslatableContext.Consumer>
+      </TranslatableProvider>
+    );
+
+    expect(contextValue.defaultLanguage).toBe(customDefaultLanguage);
+    expect(contextValue.languages).toEqual(customLanguages);
+  });
+
+  it('getSource returns the correct translation key', () => {
+    let contextValue: any;
+    render(
+      <TranslatableProvider>
+        <TranslatableContext.Consumer>
+          {(value) => {
+            contextValue = value;
+            return null;
+          }}
+        </TranslatableContext.Consumer>
+      </TranslatableProvider>
+    );
```

Comment:
This same code block is four times in this file, and there's a fifth version with some added props passed. Moving this to a function, if possible, would reduce code duplication.